src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -1485,6 +1750,97 @@ static void HandleInputEvents(SDL_Joystick *joystick)
     if (errno == ENODEV) {
         /* We have to wait until the JoystickDetect callback to remove this */
         joystick->hwdata->gone = SDL_TRUE;
+        errno = 0;
+    }
+
+    if (joystick->hwdata->report_sensor) {
+        SDL_assert(joystick->hwdata->fd_sensor >= 0);
```

Comment:
Is it worth opening/closing fd_sensor when the sensor is enabled and disabled? Steam for example leaves the sensor disabled unless there's a Steam Input binding active that uses it.